Sources/Ink/Internal/Character+Escaping.swift

```diff
@@ -3,8 +3,80 @@
 *  Copyright (c) John Sundell 2019
 *  MIT license, see LICENSE file for details
 */
+// ASCII chars escapable from CommonMark spec
+// \!\"\#\$\%\&\'\(\)\*\+\,\-\.\/\:\;\<\=\>\?\@\[\\\]\^\_\`\{\|\}\~
+// Result desired
+// !&quot;#$%&amp;'()*+,-./:;&lt;=&gt;?@[\]^_`{|}~
+// An ASCII punctuation character is !, ", #, $, %, &, ', (, ), *, +, ,, -, .,/
+// (U+0021–2F), :, ;, <, =, >, ?, @ (U+003A–0040), [, \, ], ^, _,`
+// (U+005B–0060), {, |, }, or ~ (U+007B–007E).
+let escapeSubstitutions: [Character: String] = [
+    "!": "!",
+    "#": "#",
+    "$": "$",
+    "%": "%",
+    "(": "(",
+    ")": ")",
+    "\'": "\'",
+    "*": "*",
+    "+": "+",
+    ",": ",",
+    "-": "-",
+    ".": ".",
+    "/": "/",
+    "\\": "\\",
+    ":": ":",
+    ";": ";",
+    "=": "=",
+    "?": "?",
+    "@": "@",
+    "[": "[",
+    "]": "]",
+    "^": "^",
+    "_": "_",
+    "`": "`",
+    "{": "{",
+    "}": "}",
+    "|": "|",
+    "~": "~",
+    ">": "&gt;",
+    "<": "&lt;",
+    "&": "&amp;",
+    "\"": "&quot;",
+    Character(Unicode.Scalar(0000)): String(Unicode.Scalar(UInt32(0xFFFD))!),
+    //Took some research to initialize this one. Security of null char replaced so it cannot be used to cause end of file/string problems somewhere else. I don't think this is easy to test?
+    Character(Unicode.Scalar(UInt32(0x0021))): String(Unicode.Scalar(UInt32(0x0021))!),
```

Comment:
Is it just me or does this seem a bit like overkill? Is the benefit of including this worth the effort of understanding/maintaining all of these manual unicode codes? Also seems like they are mostly `Character: String` pairs of the same exact character code?

Reply:
I reworked the  code so that an escaped punctuation causes the parser to skip over the backslash and then continue to use the "Substring of the original source strategy" for the retained punctuation. The tricky bit is to not allow the punctuation to then be parsed as a Markdown syntax character. Testcases should help this from regressing.